src/main/webapp/populateData.js

```diff
@@ -143,17 +143,36 @@ function populateCalendar() {
       })
       .then((hoursJson) => {
         // Display the days and the free hours for each one of them
-        const days = ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su'];
-        for (let day = 0; day < 5; day++) {
-          const dayContainer = document.querySelector('#day-'+day);
-          dayContainer.innerText = days[(hoursJson.startDay + day) % 7];
-          const workContainer = document.querySelector('#work-day-'+day);
-          workContainer.innerText =
-              convertTime(hoursJson.workHoursPerDay[day]) + ' (working)';
-          const personalContainer =
-              document.querySelector('#personal-day-'+day);
-          personalContainer.innerText =
-              convertTime(hoursJson.personalHoursPerDay[day]) + ' (personal)';
+        const days = ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'];
+        const panelContent = document.querySelector('#panel-content');
+        panelContent.innerHTML = '';
+        for (const day in hoursJson.workTimeFree) {
+          if (typeof day == 'string') {
```

Comment:
What does it mean if the `day` is not a string?

Reply:
I used this check as day is given as a string from the response. I did not chose to make this check, but when using a `for-in` loop, the `lint.js` check requires to make a check for the type of the variable or else the check is not passed.

Reply:
Weird, but ok.